src/Libraries/Microsoft.Extensions.AI.Abstractions/Tools/HostedMcpServerTool.cs

```diff
@@ -47,7 +71,18 @@ public HostedMcpServerTool(string serverName, Uri url)
     /// <summary>
     /// Gets the URL of the remote MCP server.
     /// </summary>
-    public Uri Url { get; }
+    /// <remarks>
+    /// This property is <see langword="null"/> when the tool is configured with a <see cref="ConnectorID" /> instead of a URL.
+    /// </remarks>
+    public Uri? Url { get; }
```

Comment:
@stephentoub sorry to bring this back to you.

> Would that work? Is it problematic? 

Supporting Connector ID will be problematic for this existing Uri prop.  We will need to replace it with a string (and rename it) so it can hold both url and connector ID.  We also have the `.ctor(string, Uri)`, that will need to go away too.

I think I picked `Uri` because that's what the few existing url properties used, see [query](https://github.com/search?q=repo%3Adotnet%2Fextensions+%22public+Uri%22+path%3Asrc%2FLibraries%2F*AI*%2F**%2F*&type=code).